### ISSUE TITLE
WrapperError not being exported correctly in build

### ISSUE THREAD

**Issue by chandru89new** [afsheen-01/vercel-ts-sdk#34]
**Issue:**
After building, I notice that the WrapperError type is being exported incorrectly (dynamic exports referencing an internal folder). This makes `error` have `any` type when using the library as an npm package.

**How the build definition file looks:**
<img width="484" alt="image" src="https://user-images.githubusercontent.com/10336112/194798607-66ea66eb-a6d7-4d2d-850b-9d40dae3a6a7.png">
(Notice the IDE showing a red wave)

As a result, the `error` has `any` type when this SDK is used as an external dependency:
<img width="509" alt="image" src="https://user-images.githubusercontent.com/10336112/194798665-82691d50-7e6b-42b0-8e16-eb875e848dac.png">

**Comment by chandru89new** [comment]
Refernces:
- https://github.com/parcel-bundler/parcel/issues/8048
- https://github.com/parcel-bundler/parcel/issues/7951

It appears to be an unresolved issue in Parcel bundler.

**Comment by chandru89new** [comment]
As recently as 2nd Sep: https://github.com/parcel-bundler/parcel/issues/8434

**Comment by chandru89new** [comment]
For now, PR #37  fixes this bug. The PR has details about how the bug was resolved.

Here's what I explored before landing on that solution:

- I checked if there are other bundlers that could keep it simple and also generate .d.ts files. Currently, some of the fastest and simplest bundlers (and transpilers) includes `esbuild`, `swc`, `vite`, and `parcel`. (Rollup does have TS support but via plugins, not out of the box).
- `vite` is focused on browsers and rapid HMR etc so it did not even seem like a possible candidate for a library like ours
- `swc` and `esbuild` both have the same issue: no `.d.ts` support yet (although both of them are unbelievably fast - and in fact, we use `swc/jest` for our tests... they helped cut down our test times by 50%)

So I decided that maybe the other way would be to bundle the JS using Parcel, but generate (and maybe later attempt to bundle the `.d.ts` via some other mechanism) using `tsc`. This is what PR #37 does.

cc: @afsheen-01
